src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -103,26 +100,33 @@ public static Date parseDate(String date) throws ParseException {
      *
      * @throws ParseException if the given {@code tag} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
+    public static Tag parseTag(EntryType type, String tag) throws ParseException {
         requireNonNull(tag);
         String trimmedTag = tag.trim();
-        if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+        if (!Tag.isValidTagName(type, trimmedTag)) {
```

Comment:
The code here seems a little confusing, if the tag is invalid, then proceed to switch case?

Reply:
Perhaps it would be better to add a guard clause instead?

Reply:
Actually, upon looking closer, I realised I made a mistake and you're already using a guard clause! Perhaps we could abstract away the exception handling part to a new function like getErrorMessagebyType(....) to make the code more readable? But it's just a suggestion, considering the code here is already clear in what it does 👍